src/main/java/io/takari/aether/connector/AetherRepositoryConnectorFactory.java

```diff
@@ -61,18 +71,137 @@ public boolean equals(Object obj) {
 
   // Default constructor required for the service locator to work with you use this factory outside the confines of Guice.
   public  AetherRepositoryConnectorFactory() throws NoSuchAlgorithmException {
-    this(null, null);
+    this(null, null, null);
   }
   
   @Inject
-  public AetherRepositoryConnectorFactory(FileProcessor fileProcessor, @Nullable SSLSocketFactory sslSocketFactory) throws NoSuchAlgorithmException {
+  public AetherRepositoryConnectorFactory(FileProcessor fileProcessor, @Nullable SSLSocketFactory sslSocketFactory, @Nullable X509TrustManager trustManager) throws NoSuchAlgorithmException {
     this.fileProcessor = fileProcessor;
     
-    // explicitly use jdk-default ssl socket factory if none is provided externally
+    // explicitly create new ssl socket factory if none is provided externally
     // this is necessary because okhttp default socket factory does not honour
     // javax.net.ssl.keyStore/trustStore system properties, which are the only way
     // to use custom ket/trust stores in maven in m2e
-    this.sslSocketFactory = sslSocketFactory != null? sslSocketFactory: SSLContext.getDefault().getSocketFactory();
+    if (sslSocketFactory != null && trustManager != null) {
+      this.sslSocketFactory = sslSocketFactory;
+      this.trustManager = trustManager;
+    } else {
+      // can not use use jdk-default ssl socket factory like previous because
+      // we can not get trust manager without reflection
+      try {
+        if (DefaultManagersHolder.reservedException != null)
+          throw DefaultManagersHolder.reservedException;
+        SSLContext sslContext = SSLContext.getInstance("SSL");
+        sslContext.init(DefaultManagersHolder.keyManagers, DefaultManagersHolder.trustManagers, null);
+        this.sslSocketFactory = sslContext.getSocketFactory();
+        this.trustManager = (X509TrustManager) DefaultManagersHolder.trustManagers[0];
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  // Copied from sun.security.ssl.SSLContextImpl
+  private static final class DefaultManagersHolder {
```

Comment:
sun.security.ssl.SSLContextImpl from OpenJDK is GPLv2. I don't believe we can include it with EPL code